File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s456888268_commented.py
import sys

def solve():
    tmp = ns()
    s = list(tmp)
    n = len(s)
    ans = 0
    cnt = 0
    for i in range(n):
        if s[i] == 'S':
            cnt += 1
        else:
            if cnt == 0:
                continue
            ans += 1
            cnt -= 1
    print(n - 2 * ans)

# constants
mod = int(1e9 + 7)
dx = [-1, 0, 1, 0]
dy = [0, -1, 0, 1]
dx8 = [-1, -1, -1, 0, 0, 1, 1, 1]
dy8 = [-1, 0, 1, -1, 1, -1, 0, 1]
inf = float('inf')
linf = float('inf')
dinf = float('inf')
eps = 1e-10
pi = 3.141592653589793
sb = []
is_input = sys.stdin
out = sys.stdout

def reverse(arr):
    if isinstance(arr, list):
        return arr[::-1]
    elif isinstance(arr, str):
        return arr[::-1]
    else:
        raise TypeError("Unsupported type for reverse")

def get_reverse(s):
    return reverse(s)

def lower_bound(a, x):
    l, r = -1, len(a)
    while r - l > 1:
        c = (l + r) // 2
        if a[c] < x:
            l = c
        else:
            r = c
    return r

def upper_bound(a, x):
    l, r = -1, len(a)
    while r - l > 1:
        c = (l + r) // 2
        if a[c] <= x:
            l = c
        else:
            r = c
    return r

def rlower_bound(a, x):
    l, r = -1, len(a)
    while r - l > 1:
        c = (l + r) // 2
        if a[c] > x:
            l = c
        else:
            r = c
    return r

def rupper_bound(a, x):
    l, r = -1, len(a)
    while r - l > 1:
        c = (l + r) // 2
        if a[c] >= x:
            l = c
        else:
            r = c
    return r

def concat(x, arr):
    if isinstance(arr, list):
        return [x] + arr
    elif isinstance(arr, str):
        return x + arr
    else:
        raise TypeError("Unsupported type for concat")

def max3(x, y, z):
    return max(x, y, z)

def min3(x, y, z):
    return min(x, y, z)

def sort(arr):
    return sorted(arr)

def rsort(arr):
    return sorted(arr, reverse=True)

def fill(arr, x):
    return [x] * len(arr)

def plus(x, y):
    res = (x + y) % mod
    return res if res >= 0 else res + mod

def sub(x, y):
    res = (x - y) % mod
    return res if res >= 0 else res + mod

def mul(x, y):
    res = (x * y) % mod
    return res if res >= 0 else res + mod

def div(x, y):
    res = x * pow(y, mod - 2, mod) % mod
    return res if res >= 0 else res + mod

def pow(x, y):
    if y < 0:
        return 0
    if y == 0:
        return 1
    if y % 2 == 1:
        return (x * pow(x, y - 1, mod)) % mod
    root = pow(x, y // 2, mod)
    return root * root % mod

def ns():
    while True:
        c = sys.stdin.read(1)
        if c == '':
            return ''
        if c in '\n \r\t':
            continue
        s = c
        while True:
            c = sys.stdin.read(1)
            if c in '\n \r\t' or c == '':
                break
            s += c
        return s

def ni():
    return int(ns())

def nl():
    return int(ns())

def nd():
    return float(ns())

def main():
    solve()

if __name__ == "__main__":
    main()
# End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/C/s456888268_commented.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <ctype.h>

#define MOD 1000000007
#define MAX 1024

int dx[] = { -1, 0, 1, 0 };
int dy[] = { 0, -1, 0, 1 };
int dx8[] = { -1, -1, -1, 0, 0, 1, 1, 1 };
int dy8[] = { -1, 0, 1, -1, 1, -1, 0, 1 };
int inf = 1073741823;
long long linf = 6372812765430547558LL;
double dinf = 1.7976931348623157E308;
double eps = 1e-10;
double pi = 3.14159265358979323846;

char inbuf[MAX];
int lenbuf = 0, ptrbuf = 0;

void solve() {
    char tmp[MAX];
    scanf("%s", tmp);
    int n = strlen(tmp);
    int ans = 0;
    int cnt = 0;
    for (int i = 0; i < n; i++) {
        if (tmp[i] == 'S') {
            cnt++;
        } else {
            if (cnt == 0) continue;
            ans++;
            cnt--;
        }
    }
    printf("%d\n", n - 2 * ans);
}

int readByte() {
    if (lenbuf == -1) return -1;
    if (ptrbuf >= lenbuf) {
        ptrbuf = 0;
        lenbuf = fread(inbuf, 1, MAX, stdin);
        if (lenbuf <= 0) return -1;
    }
    return inbuf[ptrbuf++];
}

int isSpaceChar(int c) {
    return !(c >= 33 && c <= 126);
}

int skip() {
    int b;
    while ((b = readByte()) != -1 && isSpaceChar(b));
    return b;
}

char* ns() {
    int b = skip();
    char* sb = (char*) malloc(MAX * sizeof(char));
    int p = 0;
    while (!(isSpaceChar(b))) {
        sb[p++] = (char) b;
        b = readByte();
    }
    sb[p] = '\0';
    return sb;
}

int ni() {
    int num = 0, b;
    int minus = 0;
    while ((b = readByte()) != -1 && !((b >= '0' && b <= '9') || b == '-'));
    if (b == '-') {
        minus = 1;
        b = readByte();
    }
    while (true) {
        if (b >= '0' && b <= '9') {
            num = num * 10 + (b - '0');
        } else {
            return minus ? -num : num;
        }
        b = readByte();
    }
}

int main() {
    solve();
    return 0;
}
// End of Code
